src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const array = century ? people.filter(element => element.sex === 'm'
+    && century === Math.ceil(element.died / 100))
+    : people.filter(element => element.sex === 'm');
+
+  const finalAge = century ? (array.map((element) =>
+    element.died - element.born).reduce((prev, next) => prev + next)
+    / array.length)
+    : (array.map((element) => element.died
+    - element.born).reduce((prev, next) => prev + next) / array.length);
```

Comment:
do we need `?:` here
we filter our array previously and here you do the same logic